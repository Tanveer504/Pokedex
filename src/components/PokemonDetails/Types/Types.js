import React from "react";
import "./Types.scss";

function Types({ types }) {
  return (
    <div className="PokemonDetail__Types">
      {types && types.map((type, index) => (
        <span className={type.type.name} key={index}>
          {type.type.name}
        </span>
      ))}
    </div>
  );
}

export default Types;