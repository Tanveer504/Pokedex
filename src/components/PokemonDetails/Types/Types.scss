@import '../../../styles/color.scss';
@import "../../../styles/Breakpoint.scss";

//Scss Beautifier
//mobile first approach
//types

//600 px below for mobile
.PokemonDetail__Types {
  position: absolute;
  width: 43px;
  height: 25px;
  left: 215px;
  top: 669px;
  span {
    padding: 1px;
    border: 1px solid #2e3156;
    border-radius: 5px;
    margin-right: 6px;
    text-transform: capitalize;
  }.ghost {
  background: $background-color_1;
}
.electric {
  background: $background-color_2;
}
.bug {
  background: $background-color_3;
}
.poison {
  background: $background-color_4;
}
.normal {
  background: $background-color_5;
}
.flying {
  background: $background-color_6;
}
.fairy {
  background: $background-color_7;
}
.fire {
  background: $background-color_8;
}
.grass {
  background: $background-color_9;
}
.water {
  background: $background-color_10;
}
.ground {
  background: $background-color_11;
}
.rock {
  background: $background-color_12;
}
}

@media only screen and (min-width: $min-width-desktop) {
  .PokemonDetail__Types {
    box-sizing: border-box;
    position: absolute;
    width: 39px;
    height: 24px;
    left: 245px;
    top: 500px;
    span {
      padding: 1px;
      border: 1px solid #2e3156;
      border-radius: 5px;
      margin-right: 6px;
    }
  }
}
