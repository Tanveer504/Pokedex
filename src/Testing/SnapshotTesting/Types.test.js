import renderer from 'react-test-renderer';
import Types from '../../components/PokemonDetails/Types'

it('should have the ability component snapshot', () => {
    const tree = renderer.create(<Types />).toJSON();
    expect(tree).toMatchSnapshot();
});