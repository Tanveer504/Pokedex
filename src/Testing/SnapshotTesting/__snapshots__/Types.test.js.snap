// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have the ability component snapshot 1`] = `
<div
  className="PokemonDetail__Types"
/>
`;
